packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -109,12 +145,42 @@ function compressInterval(interval: ISerializedInterval): CompressedSerializedIn
 	];
 
 	if (interval.stickiness !== undefined && interval.stickiness !== IntervalStickiness.END) {
-		base.push(interval.stickiness);
+		// reassignment to make it easier for typescript to reason about types
+		base = [
+			...base,
+			interval.stickiness,
+			interval.startSide ?? Side.Before,
+			interval.endSide ?? Side.After,
+		];
 	}
 
 	return base;
 }
 
+/**
+ * @internal
+ */
+export function endpointPosAndSide(
+	start: number | "start" | "end" | SequencePlace | undefined,
+	end: number | "start" | "end" | SequencePlace | undefined,
+) {
+	const startIsPlainEndpoint = typeof start === "number" || start === "start" || start === "end";
+	const endIsPlainEndpoint = typeof end === "number" || end === "start" || end === "end";
+
+	const startSide = startIsPlainEndpoint ? Side.Before : start?.side;
+	const endSide = endIsPlainEndpoint ? Side.After : end?.side;
```

Comment:
I had been thinking that currently our intervals are technically inclusive of the endpoint (even if semantically, some partners treat it exclusively) and so defaulting to an exclusive endpoint would result in different behavior.

Reply:
without any stickiness involved / historically, interval endpoints shift forward whenever available. This would give an app interpreting the end endpoint as having really buggy behavior when a segment containing its former end gets deleted--say the endpoints are on "A" and "C" of "ABCDE" and "BCD" gets deleted. If you interpret the interval inclusively as an application, you'd suddenly see "E" as part of your interval which doesn't seem right. If you interpret the interval exclusively, things generally work out ok (but boundary conditions get wonky because `forward ?? backward ?? detach` doesn't really make sense under any scheme)